app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -24,6 +24,9 @@ def set_request_start
 
   # user auth
   def set_user_by_token(mapping = nil)
+    # Avoid checking resources multi times
+    return @resource if resource_is_checked?(mapping)
```

Comment:
should we just use `defined?(@resource)` instead of this elaborate check?

Reply:
Agreed

Reply:
I'm try set `return @resource if defined?(@resource)`, It works fine when logged in. 
But, when not logged in it still checks resources many times.
![Screenshot from 2021-01-28 10-49-23](https://user-images.githubusercontent.com/6843034/106087798-53264c80-6157-11eb-9eaa-1a4d2465e63d.png)